src/doc/src/reference/resolver.md

```diff
@@ -0,0 +1,415 @@
+`1.0.0-alpha.4` will use numeric comparison for the `4` component. That means
+that if `1.0.0-alpha.11` is published, that will be chosen as the greatest
+release. Non-numeric components are compared lexicographically.
+
+[`cargo install`]: ../commands/cargo-install.md
+
+### Version metadata
+
+SemVer has the concept of "version metadata" with a plus in the version, such
+as `1.0.0+21AF26D3`. This metadata is usually ignored, and should not be used
+in a version requirement. You should never publish multiple versions that
+differ only in the metadata tag (note, this is a [known issue] with
+[crates.io] that currently permits this).
+
+[known issue]: https://github.com/rust-lang/crates.io/issues/1059
+[crates.io]: https://crates.io/
+
+## Other constraints
+
+Version requirements aren't the only constraint that the resolver considers
+when selecting and unifying dependencies. The following sections cover some of
+the other constraints that can affect resolution.
+
+### Features
+
+The resolver resolves the graph as-if the [features] of all [workspace]
+members are enabled. This ensures that any optional dependencies are available
+and properly resolved with the rest of the graph when features are added or
+removed with the `--features` command-line flag. The actual features used when
+*compiling* a crate will depend on the features enabled on the command-line.
+
+Dependencies are resolved with the union of all features enabled on them. For
+example, if one package depends on the [`im`] package with the [`serde`
+dependency] enabled and another package depends on it with the [`rayon`
+dependency] enabled, then `im` will be built with both features enabled, and
+the `serde` and `rayon` crates will be included in the resolve graph. If no
+packages depend on `im` with those features, then those optional dependencies
+will be ignored, and they will not affect resolution.
+
+The resolver will skip over versions of packages that are missing required
+features. For example, if a package depends on version `^1` of [`regex`] with
+the [`perf` feature], then the oldest version it can select is `1.3.0`,
+because versions prior to that did not contain the `perf` feature. Similarly,
+if a feature is removed from a new release, then packages that require that
+feature will be stuck on the older releases that contain that feature. It is
+discouraged to remove features in a SemVer-compatible release.
+
+[`im`]: https://crates.io/crates/im
+[`perf` feature]: https://github.com/rust-lang/regex/blob/1.3.0/Cargo.toml#L56
+[`rayon` dependency]: https://github.com/bodil/im-rs/blob/v15.0.0/Cargo.toml#L47
+[`regex`]: https://crates.io/crates/regex
+[`serde` dependency]: https://github.com/bodil/im-rs/blob/v15.0.0/Cargo.toml#L46
+[features]: features.md
+[workspace]: workspaces.md
+
+### `links`
+
+The [`links` field] is used to ensure only one copy of a native library is
+linked into a binary. It is an error if multiple SemVer-incompatible versions
```

Comment:
This is all true, but I feel the tone is a little harsh. The resolver will attempt to find a graph where there is only one, and only throw an error if there is no graph that works. Above you used the phrase "will fail because" for the analogous situation with SemVer-compatible versions, where here it is introduced with "is an error".

Reply:
I tried to tweak the wording a bit, let me know if that works.